src/github.rs

```diff
@@ -1,28 +1,25 @@
 //! This defines the GitHub RSS feed and API, to get the commits of the
 //! repository (among other data).
 
-use crate::prelude::*;
-use once_cell::sync::Lazy;
-use std::env;
-
-pub static GITHUB: Lazy<Github> = Lazy::new(|| {
-    let repository =
-        env::var("GITHUB_REPOSITORY").unwrap_or_else(|_| String::from("PaperMC/Paper"));
-
-    info!("Using Github repository: {}", repository);
-
-    Github { repository }
-});
-
-static DOMAIN: Lazy<String> = Lazy::new(|| {
-    env::var("GITHUB_DOMAIN").unwrap_or_else(|_| String::from("https://www.github.com"))
-});
-static API_DOMAIN: Lazy<String> = Lazy::new(|| {
-    env::var("GITHUB_API_DOMAIN").unwrap_or_else(|_| String::from("https://api.github.com"))
-});
+const ACCEPT_ATOM: &'static str = "application/atom+xml";
 
 pub struct Github {
     repository: String,
 }
 
-impl Github {}
+impl Default for Github {
+    fn default() -> Self {
+        Self::new("PaperMC/Paper")
+    }
+}
+
+impl Github {
+    pub fn new<T>(repository: T) -> Self
+    where
+        T: Into<String>,
```

Comment:
Time for code tricks!
```suggestion
    pub fn new(repository: impl Into<String>) -> Self
```

Though perhaps we should consider using `impl Into<Cow<'a, str>>` or something? Chances are we can accept a that in the "main" method of the lib, which would mean we could avoid an allocation extra. The following compiles after all:

```rust
let cow: Cow<str> = "hello".into(); // => impl<'a> From<&'a str> for Cow<'a, str>
let cow: Cow<str> = String::from("hello").into(); // => impl<'a> From<String> for Cow<'a, str>
```

Reply:
I'm not sure we want to go all the way to Cow, especially given that we do store it in the struct. Do we really want to replace (almost) all of our `String`s with `Cow`s?